doc: matter: Added ZMS mentions to the Matter docs (#17952)
nrfconnect/sdk-nrf: doc/nrf/protocols/matter/end_product/bootloader.rst

This requires that you define the ``settings_storage`` partition in the flash.
The recommended minimum size of the partition is 32 kB, but you can reserve even more space if your application uses the storage extensively.

The Zephyr settings storage is implemented by the :ref:`Zephyr NVS (Non-Volatile Storage) <zephyr:nvs_api>` or :ref:`ZMS (Zephyr Memory Storage) <zephyr:zms_api>` backends.

> The backend selection is up to you, however it makes implications to the many factors, like the operations performance or memory life-time.

Reviewer: ```suggestion
The backend selection is up to you, however, it has implications for many factors, such as operational performance and memory lifetime.
To achieve the optimal experience, it is recommended to use:
```

To achieve the optimal experience, it is recommended to use:

* NVS backend for the flash based nRF52 and nRF53 SoC families.

> * ZMS backend for the RRAM and MRAM based nRF54 SoC families.

Reviewer 2: ```suggestion
* ZMS backend for the RRAM- and MRAM-based nRF54 SoC families.
```

> The settings backend uses multiple sectors of 4 kB each and it has to use the appropriate number of sectors to cover all settings partition area.

Reviewer: ```suggestion
The settings backend uses multiple sectors of 4 kB each, and it must use the appropriate number of sectors to cover the entire settings partition area.
```

> To configure the number of sectors used by the backend, set the following Kconfig option to the desired value:

Reviewer 2: ```suggestion
To configure the number of sectors used by the backend, set the corresponding Kconfig option to the desired value:
```
